DataFormats/PatCandidates/src/TriggerObjectStandAlone.cc

```diff
@@ -306,3 +328,120 @@ void TriggerObjectStandAlone::unpackPathNames(const edm::TriggerNames &names) {
+        }
+    }
+    std::sort(indices.begin(), indices.end()); // try reduce enthropy
+    filterLabelIndices_.swap(indices);
+    filterLabels_.swap(unmatched);
+}
+
+void TriggerObjectStandAlone::unpackFilterLabels() const {
+  unpackFilterLabels(*allLabels(psetId_));
+}
+
+void TriggerObjectStandAlone::unpackFilterLabels(const std::vector<std::string> &labels) const {
+    if (filterLabelIndices_.empty()) return;
+
+    std::vector<std::string> mylabels(filterLabels_);
+    for (unsigned int i = 0, n = filterLabelIndices_.size(), m = labels.size(); i < n; ++i) {
+        if (filterLabelIndices_[i] >= m) throw cms::Exception("RuntimeError", "Error, filter label index out of bounds");
+        mylabels.push_back(labels[i]);
+    }
+    filterLabelIndices_.clear();
+    filterLabels_.swap(mylabels);
+}
+
+void TriggerObjectStandAlone::packP4() {
+    setP4(reco::Particle::PolarLorentzVector(
+                MiniFloatConverter::reduceMantissaToNbitsRounding<14>(pt()),
+                MiniFloatConverter::reduceMantissaToNbitsRounding<11>(eta()),
+                MiniFloatConverter::reduceMantissaToNbits<11>(phi()),
+                MiniFloatConverter::reduceMantissaToNbitsRounding<8>(mass()) ));
+}
+
+namespace {
+ struct key_hash {
+          std::size_t operator()(edm::ParameterSetID const& iKey) const{
+             return iKey.smallHash();
+          }
+       };
+  typedef tbb::concurrent_unordered_map<edm::ParameterSetID, std::vector<std::string>, key_hash> AllLabelsMap;
+  [[cms::thread_safe]] static AllLabelsMap allLabelsMap;
+}
+
+/*std::vector<std::string>  const* TriggerObjectStandAlone::allLabels(edm::TriggerResults const& triggerResults,
+						edm::ProcessHistory const &history,std::string const& processName) {
+triggerResults.parameterSetID()
+}*/
+
+std::vector<std::string>  const* TriggerObjectStandAlone::allLabels(edm::ParameterSetID const& psetid) const{
+
+      // If TriggerNames was already created and cached here in the map,
+      // then look it up and return that one
+      AllLabelsMap::const_iterator iter =
+         allLabelsMap.find(psetid);
+      if (iter != allLabelsMap.end()) {
+         return &iter->second;
+      }
+
+      // Look for the parameter set containing the trigger names in the parameter
+      // set registry using the ID from TriggerResults as the key used to find it.
+    //  edm::ProcessConfiguration config;
+     // history.getConfigurationForProcess(processName,config); 
+	
+      //if (0!=(pset=psetRegistry->getMapped(config.parameterSetID()))) {
+
+      edm::pset::Registry* psetRegistry = edm::pset::Registry::instance();
+      edm::ParameterSet const* pset=0;
+      if (0!=(pset=psetRegistry->getMapped(psetid ))) {
+   	using namespace std;
+	using namespace edm;
+   	using namespace trigger;
+
+   	const ParameterSet& HLTPSet(pset->getParameterSet("@trigger_paths"));
+   	auto   triggerNames= HLTPSet.getParameter<vector<string> >("@trigger_paths");
```

Comment:
this is taken from here: 
https://github.com/cms-sw/cmssw/blob/master/HLTrigger/HLTcore/src/HLTConfigData.cc#L91-L95